blockdag/reachability.go

```diff
@@ -665,125 +671,127 @@ func (rtn *reachabilityTreeNode) String() string {
-			// block is an ancestor of candidate, and can thus replace it
-			(*fb)[i-1] = block
+			// node is an ancestor of candidate, and can thus replace it
+			(*fb)[ancestorIndex] = node
 			return
 		}
 	}
 
-	// Insert block in the correct index to maintain futureCoveringBlockSet as
+	// Insert node in the correct index to maintain futureCoveringTreeNodeSet as
 	// a sorted-by-interval list.
-	// Note that i might be equal to len(futureCoveringBlockSet)
-	left := (*fb)[:i]
-	right := append([]*futureCoveringBlock{block}, (*fb)[i:]...)
+	// Note that ancestorIndex might be equal to len(futureCoveringTreeNodeSet)
+	left := (*fb)[:ancestorIndex+1]
+	right := append([]*reachabilityTreeNode{node}, (*fb)[ancestorIndex+1:]...)
 	*fb = append(left, right...)
 }
 
-// isInFuture resolves whether the given block is in the subtree of
-// any block in this futureCoveringBlockSet.
-// See insertBlock method for the complementary insertion behavior.
+// isInFuture resolves whether the given node is in the subtree of
+// any node in this futureCoveringTreeNodeSet.
+// See insertNode method for the complementary insertion behavior.
 //
 // Like the insert method, this method also relies on the fact that
-// futureCoveringBlockSet is kept ordered by interval to efficiently perform a
-// binary search over futureCoveringBlockSet and answer the query in
-// O(log(|futureCoveringBlockSet|)).
-func (fb futureCoveringBlockSet) isInFuture(block *futureCoveringBlock) bool {
-	i := fb.findIndex(block)
-	if i == 0 {
-		// No candidate to contain block
+// futureCoveringTreeNodeSet is kept ordered by interval to efficiently perform a
+// binary search over futureCoveringTreeNodeSet and answer the query in
+// O(log(|futureCoveringTreeNodeSet|)).
+func (fb futureCoveringTreeNodeSet) isInFuture(node *reachabilityTreeNode) bool {
+	ancestorIndex, ok := treeNodeSet(fb).findAncestorIndexOfNode(node)
+	if !ok {
+		// No candidate to contain node
 		return false
 	}
 
-	candidate := fb[i-1]
-	return candidate.treeNode.isAncestorOf(block.treeNode)
+	candidate := fb[ancestorIndex]
+	return candidate.isAncestorOf(node)
+}
+
+// findAncestorOfNode finds the reachability tree ancestor of `node`
+// among the nodes in `tns`.
+func (tns treeNodeSet) findAncestorOfNode(node *reachabilityTreeNode) (*reachabilityTreeNode, bool) {
+	ancestorIndex, ok := tns.findAncestorIndexOfNode(node)
+	if !ok {
+		return nil, false
+	}
+	return tns[ancestorIndex], true
 }
 
-// findIndex finds the index of the block with the maximum start that is below
-// the given block.
-func (fb futureCoveringBlockSet) findIndex(block *futureCoveringBlock) int {
-	blockInterval := block.treeNode.interval
+// findAncestorIndexOfNode finds the index of the reachability tree
+// ancestor of `node` among the nodes in `tns`. It does so by finding
+// the index of the block with the maximum start that is below the
+// given block.
+func (tns treeNodeSet) findAncestorIndexOfNode(node *reachabilityTreeNode) (int, bool) {
```

Comment:
please return uint

Reply:
This function explicitly returns an index, which must be int.